app/controllers/course/users_controller.rb

```diff
@@ -6,15 +6,15 @@ class Course::UsersController < Course::ComponentController
   before_action :ensure_unregistered_user, only: [:create, :register]
 
   def students # :nodoc:
-    @course_users = @course_users.students.with_approved_state
+    @course_users = @course_users.students.with_approved_state.includes(:user, user: :emails)
   end
 
   def staff # :nodoc:
-    @course_users = @course_users.staff
+    @course_users = @course_users.staff.includes(:user, user: :emails)
   end
 
   def requests # :nodoc:
-    @course_users = @course_users.with_requested_state
+    @course_users = @course_users.with_requested_state.includes(:user, user: :emails)
```

Comment:
I think the first `user` is redundant ?  `includes(user: :emails)` would be enough